src/MSBuildBinLogQuery/Token/ErrorToken.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Build.Logging.Query.Token
+{
+    public class ErrorToken : Token, IEquatable<ErrorToken>
+    {
+        public ErrorToken() : base()
+        {
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as ErrorToken);
+        }
+
+        public bool Equals([AllowNull] ErrorToken other)
+        {
+            return other != null;
```

Comment:
Could this repetitive implementation be simplified by adding a new `ConstantToken` class that does the comparisons?

Reply:
Do you mean an `enum` field or a string field inside `ConstantToken`? I still prefer to have `MessageToken`, `ErrorToken`, etc. separately because that provides more compile-time verifications.

Reply:
No, I like separate classes, too. Just wondering if they could have a common base type to do the identical parts of their implementation.

Reply:
Would it be hard because they need to implement `IEquatable` and hence implement `Equals` with different parameter types?

Reply:
I like the singleton option better as it's much more straight forward, but for the record, you could use CRTP to solve that part. https://stackoverflow.com/a/2032759